tests/test_merge.py

```diff
@@ -484,3 +486,50 @@ def test_merge_allow_objects() -> None:
     cfg._set_flag("allow_objects", True)
     ret = OmegaConf.merge(cfg, {"foo": iv})
     assert ret == {"a": 10, "foo": iv}
+
+
+@pytest.mark.parametrize(  # type:ignore
+    "dst, other, expected, node, check_missing",
+    [
+        pytest.param(
+            OmegaConf.structured(InterpolationList),
+            OmegaConf.create({"list": [0.1]}),
+            {"list": [0.1]},
+            "list",
+            False,
+            id="merge_interpolation_list_with_list",
+        ),
+        pytest.param(
+            OmegaConf.structured(InterpolationDict),
+            OmegaConf.create({"dict": {"a": 4}}),
+            {"dict": {"a": 4}},
+            "dict",
+            False,
+            id="merge_interpolation_dict_with_dict",
+        ),
+        pytest.param(
+            OmegaConf.structured(InterpolationDict),
+            OmegaConf.structured(InterpolationDict),
+            None,
+            "dict",
+            True,
+            id="merge_interpolation_dict_with_interpolation_dict",
+        ),
+        pytest.param(
+            OmegaConf.structured(InterpolationList),
+            OmegaConf.structured(InterpolationList),
+            None,
+            "list",
+            True,
+            id="merge_interpolation_list_with_interpolation_list",
+        ),
+    ],
+)
+def test_merge_with_interpolation(
+    dst: Any, other: Any, expected: Any, node: Any, check_missing: bool
+) -> None:
+    res = OmegaConf.merge(dst, other)
+    if check_missing:
+        OmegaConf.is_missing(res, node)
+    else:
+        assert res == expected
```

Comment:
do one thing in each test.
you can add another test to test the behavior of _is_missing with interpolation.

Reply:
I'm changing it to is_interpolation